python/TagAndProbeModel.py

```diff
@@ -0,0 +1,43 @@
+from HiggsAnalysis.CombinedLimit.PhysicsModel import *
+import re
+
+class TagAndProbe(PhysicsModel):
+    def setPhysicsOptions(self,physOptions):
+        for po in physOptions:
+            if po.startswith("anticorrelation="):
+                self.anticorrelation = float(po.replace("anticorrelation=",""))
+    def doParametersOfInterest(self):
+        """Create POI and other parameters, and define the POI set."""
+        self.modelBuilder.doVar("SF[1.0,0.0,1.2]");
+        self.modelBuilder.doSet("POI","SF")
+        if self.options.mass != 0:
+            if self.modelBuilder.out.var("MH"):
+              self.modelBuilder.out.var("MH").removeRange()
+              self.modelBuilder.out.var("MH").setVal(self.options.mass)
+            else:
+              self.modelBuilder.doVar("MH[%g]" % self.options.mass);
+        exp_pass=1
+        exp_fail=1
+        for b in self.DC.bins:
+           for p in self.DC.exp[b].keys():
+              if self.DC.isSignal[p]:
+                 if re.search("pass",b):
+                    exp_pass=self.DC.exp[b][p]
+                 if re.search("fail",b):
+                    exp_fail=self.DC.exp[b][p]
+        self.modelBuilder.factory_('expr::fail_scale("(%f+%f-(%f*@0))/%f", SF)'%(exp_pass,exp_fail,exp_pass,exp_fail))
```

Comment:
I'm not totally convinced of the math here, in case one has additional systematics which may cause the yield to change wrt `DC.exp[b][p]` (plus, it fails if you fit two bins at once)

A simpler implementation could be to fill the signal expectation with the full yield both in the passing and in the failing category (i.e. as if the efficiency were 100% and 0% respectively) and then return eff and (1-eff) in the two. That way if you had e.g. a normalization systematic scaling the signal yield up or down by 10% you would still get the right answer.